Traceback (most recent call last):
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukem/anaconda3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/lukem/anaconda3/lib/python3.11/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def get_product_type(product):
    if product[0:2] == 'S1':
        type = product.split('_')[2]
    elif product[0:2] == 'S2':
        type = product.split('_')[1]
        if not type.startswith('M'):
            type = 'Unknown'
    elif product[0:2] == 'S3':
        tmp = product.split('_')
        if tmp[1] == 'SL':
            type = 'SLSTR_L' + tmp[2]
        elif tmp[1] == 'SR':
            type = 'SRAL_L' + tmp[2]
        elif tmp[1] == 'OL':
            type = 'OLCI_L' + tmp[2]
        else:
            type = 'Unknown'
    else:
        type = 'Unknown'
    if 'DTERRENG' in product:
        type = type + '_DTERRENG'
    return type

def get_data(file):
    data = pd.read_csv(file, header=None, names=['download_time', 'user', 'product', 'size', 'download_duration']\
                        , parse_dates=['download_time'], index_col='download_time')
    data['satellite'] = data['product'].apply(lambda x: x[0:2])
    data['product_type'] = data['product'].apply(get_product_type)
    return data[data['product_type'] != 'Unknown']

csvfile = logsdir / 'NBS_frontend-global_outputs.csv'
nbs_global = get_data(csvfile)
#nbs_global.set_index('download_time', inplace=True)
nbs_global.sort_index(inplace=True)
nbs_global.fillna(0, inplace=True)
nbs_30 = nbs_global.loc[nbs_global.index >= dt.datetime.today() - dt.timedelta(days=30)].copy()

col_use = len(nbs_30['user'].unique())
col_nb = nbs_30.count()['product_type']
col_vol = nbs_30.sum()['size']/1024/1024/1024/1024 #in Tb


csvfile = logsdir / 'NBS_frontend-AOI_outputs.csv'
nbs_AOI = get_data(csvfile)
#nbs_AOI.set_index('download_time', inplace=True)
nbs_AOI.sort_index(inplace=True)
nbs_AOI.fillna(0, inplace=True)
aoi_30 = nbs_AOI.loc[nbs_AOI.index >= dt.datetime.today() - dt.timedelta(days=30)].copy()

arc_use = len(aoi_30['user'].unique())
arc_nb = aoi_30.count()['product_type']
arc_vol = aoi_30.sum()['size']/1024/1024/1024/1024 #in Tb
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mParserError[0m                               Traceback (most recent call last)
Cell [0;32mIn[3], line 44[0m
[1;32m     40[0m col_vol [38;5;241m=[39m nbs_30[38;5;241m.[39msum()[[38;5;124m'[39m[38;5;124msize[39m[38;5;124m'[39m][38;5;241m/[39m[38;5;241m1024[39m[38;5;241m/[39m[38;5;241m1024[39m[38;5;241m/[39m[38;5;241m1024[39m[38;5;241m/[39m[38;5;241m1024[39m [38;5;66;03m#in Tb[39;00m
[1;32m     43[0m csvfile [38;5;241m=[39m logsdir [38;5;241m/[39m [38;5;124m'[39m[38;5;124mNBS_frontend-AOI_outputs.csv[39m[38;5;124m'[39m
[0;32m---> 44[0m nbs_AOI [38;5;241m=[39m get_data(csvfile)
[1;32m     45[0m [38;5;66;03m#nbs_AOI.set_index('download_time', inplace=True)[39;00m
[1;32m     46[0m nbs_AOI[38;5;241m.[39msort_index(inplace[38;5;241m=[39m[38;5;28;01mTrue[39;00m)

Cell [0;32mIn[3], line 25[0m, in [0;36mget_data[0;34m(file)[0m
[1;32m     24[0m [38;5;28;01mdef[39;00m [38;5;21mget_data[39m(file):
[0;32m---> 25[0m     data [38;5;241m=[39m pd[38;5;241m.[39mread_csv(file, header[38;5;241m=[39m[38;5;28;01mNone[39;00m, names[38;5;241m=[39m[[38;5;124m'[39m[38;5;124mdownload_time[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124muser[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mproduct[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124msize[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mdownload_duration[39m[38;5;124m'[39m]\
[1;32m     26[0m                         , parse_dates[38;5;241m=[39m[[38;5;124m'[39m[38;5;124mdownload_time[39m[38;5;124m'[39m], index_col[38;5;241m=[39m[38;5;124m'[39m[38;5;124mdownload_time[39m[38;5;124m'[39m)
[1;32m     27[0m     data[[38;5;124m'[39m[38;5;124msatellite[39m[38;5;124m'[39m] [38;5;241m=[39m data[[38;5;124m'[39m[38;5;124mproduct[39m[38;5;124m'[39m][38;5;241m.[39mapply([38;5;28;01mlambda[39;00m x: x[[38;5;241m0[39m:[38;5;241m2[39m])
[1;32m     28[0m     data[[38;5;124m'[39m[38;5;124mproduct_type[39m[38;5;124m'[39m] [38;5;241m=[39m data[[38;5;124m'[39m[38;5;124mproduct[39m[38;5;124m'[39m][38;5;241m.[39mapply(get_product_type)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/util/_decorators.py:211[0m, in [0;36mdeprecate_kwarg.<locals>._deprecate_kwarg.<locals>.wrapper[0;34m(*args, **kwargs)[0m
[1;32m    209[0m     [38;5;28;01melse[39;00m:
[1;32m    210[0m         kwargs[new_arg_name] [38;5;241m=[39m new_arg_value
[0;32m--> 211[0m [38;5;28;01mreturn[39;00m func([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/util/_decorators.py:331[0m, in [0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper[0;34m(*args, **kwargs)[0m
[1;32m    325[0m [38;5;28;01mif[39;00m [38;5;28mlen[39m(args) [38;5;241m>[39m num_allow_args:
[1;32m    326[0m     warnings[38;5;241m.[39mwarn(
[1;32m    327[0m         msg[38;5;241m.[39mformat(arguments[38;5;241m=[39m_format_argument_list(allow_args)),
[1;32m    328[0m         [38;5;167;01mFutureWarning[39;00m,
[1;32m    329[0m         stacklevel[38;5;241m=[39mfind_stack_level(),
[1;32m    330[0m     )
[0;32m--> 331[0m [38;5;28;01mreturn[39;00m func([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/io/parsers/readers.py:950[0m, in [0;36mread_csv[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)[0m
[1;32m    935[0m kwds_defaults [38;5;241m=[39m _refine_defaults_read(
[1;32m    936[0m     dialect,
[1;32m    937[0m     delimiter,
[0;32m   (...)[0m
[1;32m    946[0m     defaults[38;5;241m=[39m{[38;5;124m"[39m[38;5;124mdelimiter[39m[38;5;124m"[39m: [38;5;124m"[39m[38;5;124m,[39m[38;5;124m"[39m},
[1;32m    947[0m )
[1;32m    948[0m kwds[38;5;241m.[39mupdate(kwds_defaults)
[0;32m--> 950[0m [38;5;28;01mreturn[39;00m _read(filepath_or_buffer, kwds)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/io/parsers/readers.py:611[0m, in [0;36m_read[0;34m(filepath_or_buffer, kwds)[0m
[1;32m    608[0m     [38;5;28;01mreturn[39;00m parser
[1;32m    610[0m [38;5;28;01mwith[39;00m parser:
[0;32m--> 611[0m     [38;5;28;01mreturn[39;00m parser[38;5;241m.[39mread(nrows)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/io/parsers/readers.py:1778[0m, in [0;36mTextFileReader.read[0;34m(self, nrows)[0m
[1;32m   1771[0m nrows [38;5;241m=[39m validate_integer([38;5;124m"[39m[38;5;124mnrows[39m[38;5;124m"[39m, nrows)
[1;32m   1772[0m [38;5;28;01mtry[39;00m:
[1;32m   1773[0m     [38;5;66;03m# error: "ParserBase" has no attribute "read"[39;00m
[1;32m   1774[0m     (
[1;32m   1775[0m         index,
[1;32m   1776[0m         columns,
[1;32m   1777[0m         col_dict,
[0;32m-> 1778[0m     ) [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_engine[38;5;241m.[39mread(  [38;5;66;03m# type: ignore[attr-defined][39;00m
[1;32m   1779[0m         nrows
[1;32m   1780[0m     )
[1;32m   1781[0m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[1;32m   1782[0m     [38;5;28mself[39m[38;5;241m.[39mclose()

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/io/parsers/c_parser_wrapper.py:230[0m, in [0;36mCParserWrapper.read[0;34m(self, nrows)[0m
[1;32m    228[0m [38;5;28;01mtry[39;00m:
[1;32m    229[0m     [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mlow_memory:
[0;32m--> 230[0m         chunks [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_reader[38;5;241m.[39mread_low_memory(nrows)
[1;32m    231[0m         [38;5;66;03m# destructive to chunks[39;00m
[1;32m    232[0m         data [38;5;241m=[39m _concatenate_chunks(chunks)

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/_libs/parsers.pyx:808[0m, in [0;36mpandas._libs.parsers.TextReader.read_low_memory[0;34m()[0m

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/_libs/parsers.pyx:866[0m, in [0;36mpandas._libs.parsers.TextReader._read_rows[0;34m()[0m

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/_libs/parsers.pyx:852[0m, in [0;36mpandas._libs.parsers.TextReader._tokenize_rows[0;34m()[0m

File [0;32m~/anaconda3/lib/python3.11/site-packages/pandas/_libs/parsers.pyx:1973[0m, in [0;36mpandas._libs.parsers.raise_parser_error[0;34m()[0m

[0;31mParserError[0m: Error tokenizing data. C error: Expected 5 fields in line 248551, saw 7

ParserError: Error tokenizing data. C error: Expected 5 fields in line 248551, saw 7


